pkg/api/types.go

```diff
@@ -1509,10 +1513,26 @@ type NodeList struct {
 	Items []Node `json:"items"`
 }
 
-// NamespaceSpec describes the attributes on a Namespace
+// NamespaceNetworkPolicy determines who is authorized to access pods in the namespace
+type NamespaceNetworkPolicy string
+
+// These are the valid network policies of a namespace
+const (
+	// Closed namespaces are only accessible by pods within the namespace
+	NamespaceNetworkPolicyClosed NamespaceNetworkPolicy = "Closed"
+	// Open namespaces are accessible from any namespace
+	NamespaceNetworkPolicyOpen NamespaceNetworkPolicy = "Open"
+)
+
+// NamespaceSpec describes the attributes on a Namespace.
 type NamespaceSpec struct {
-	// Finalizers is an opaque list of values that must be empty to permanently remove object from storage
-	Finalizers []FinalizerName
+	// Finalizers is an opaque list of values that must be empty to permanently remove object from storage.
+	// More info: http://releases.k8s.io/HEAD/docs/design/namespaces.md#finalizers
+	Finalizers []FinalizerName `json:"finalizers,omitempty"`
+
+	// NetworkPolicy indicates who is authorized to access pods in the namespace.
+	// Must be either Open or Closed. Defaults to Open.
+	NetworkPolicy NamespaceNetworkPolicy `json:"experimentalNetworkPolicy,omitempty"`
```

Comment:
>So far, we have not added policies directly to namespace. If this is experimental, why not use an annotation?

Yes, I suppose this does seem like a good fit for annotations.  We first tried this out ourselves using annotations + Calico.  I guess the issue is that the only work required here is the API itself - realizing that API is up to the network provider, and shouldn't be a part of core Kubernetes.

> is there a way to configure a subset of namespaces as always being available?

This defaults namespaces as being available, meaning kube-system, default, openshift-infra, etc will all be accessible unless declared otherwise.

> Also think this should be a separate type that is added to a namespace
versus a field on the namespace.  Annotation on namespace is fine for
experimentation, but not final API.

Yes, I see what you mean.  I'm currently working on putting together a more detailed proposal for how network policy should look in Kubernetes that will try to spell out things like this a bit better.